heft-plugins/heft-api-extractor-plugin/src/ApiExtractorPlugin.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type * as TApiExtractor from '@microsoft/api-extractor';
+import type {
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IHeftTaskSession,
+  HeftConfiguration,
+  IHeftTaskCleanHookOptions
+} from '@rushstack/heft';
+import { ConfigurationFile } from '@rushstack/heft-config-file';
+
+import { ApiExtractorRunner } from './ApiExtractorRunner';
+
+const PLUGIN_NAME: string = 'ApiExtractorPlugin';
+const TASK_CONFIG_SCHEMA_PATH: string = `${__dirname}/schemas/api-extractor-task.schema.json`;
+const TASK_CONFIG_RELATIVE_PATH: string = './config/api-extractor-task.json';
+const EXTRACTOR_CONFIG_FILENAME: typeof TApiExtractor.ExtractorConfig.FILENAME = 'api-extractor.json';
+const EXTRACTOR_CONFIG_RELATIVE_PATH: string = `./config/${EXTRACTOR_CONFIG_FILENAME}`;
```

Comment:
I think technically API Extractor lets you stick the config file in the project root. It's probably fine for Heft to not support that, but we may want to warn if it's there.

Reply:
Given that it's unsupported and we've never supported it for the plugin, I'm going to make it `emitError` and gracefully exit. At least it'll provide a helpful info message for people converting to Heft that used API Extractor in the past.

Reply:
Ehh, actually gonna make this a warning. It'll still prevent checkin, but can function out of the gate. There's a few other places that this happens as well (API Extractor supported version for example).